### PR TITLE
Fix missing Keras variables shim

### PR DESCRIPTION
**Pull request by nvcastet**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/GOVERNANCE.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: horovod/_keras/__init__.py [hunk 1]
```diff
@@ -79,13 +79,15 @@ def __init__(self, **kwargs):
                         scale_local_gradients=scale_local_gradients
                     )
 
-        def variables(self):
-            if _IS_TF2:
-                return super(self.__class__, self).variables()
-            return self.get_weights()
-
         if version.parse(tf.__version__) >= version.parse('2.12.0'):
-            variables = property(variables)
```
Comment (maxhgerlach) [inline horovod/_keras/__init__.py:88]: Did this not work right in practice? (I only looked at unit tests..)

**Review by maxhgerlach — APPROVED** [review]

File: horovod/_keras/__init__.py (hunk 1 above)
Comment (maxhgerlach) [reply horovod/_keras/__init__.py:88]: I guess we didn't get the `opt.variables()` compatibility for the super class `variables()`..

**Comment by maxhgerlach** [comment]
Triggered GPU pipelines to run again. Previous failures look like flakiness. Let's wait for those to pass..

**Comment by maxhgerlach** [comment]
This time the GPU head pipeline mostly passed, except for Spark Lightning MNIST, which has been failing there for a while.

Re-triggered the regular GPU pipeline once more.

**Comment by maxhgerlach** [comment]
Hmm, it looks as if `download.pytorch.org` is not very reliable at the moment.

> Looking in links: https://download.pytorch.org/whl/cu101/torch_stable.html
> WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba0576c310>: Failed to establish a new connection: [Errno -2] Name or service not known')': /whl/cu101/torch_stable.html
> WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba0574ad90>: Failed to establish a new connection: [Errno -2] Name or service not known')': /whl/cu101/torch_stable.html
> WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba0576c4d0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /whl/cu101/torch_stable.html
> WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba0576c810>: Failed to establish a new connection: [Errno -2] Name or service not known')': /whl/cu101/torch_stable.html
> WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fba0576cb50>: Failed to establish a new connection: [Errno -2] Name or service not known')': /whl/cu101/torch_stable.html
> ERROR: Could not find a version that satisfies the requirement torch==1.5.0+cu101 (from versions: 1.0.0, 1.0.1, 1.0.1.post2, 1.1.0, 1.2.0, 1.3.0, 1.3.1, 1.4.0, 1.5.0, 1.5.1, 1.6.0, 1.7.0, 1.7.1, 1.8.0, 1.8.1, 1.9.0, 1.9.1, 1.10.0, 1.10.1, 1.10.2, 1.11.0, 1.12.0, 1.12.1, 1.13.0, 1.13.1)
> ERROR: No matching distribution found for torch==1.5.0+cu101

Some builds fail with errors like this one.

File: horovod/_keras/__init__.py (hunk 1 above)
Comment (nvcastet) [reply horovod/_keras/__init__.py:88]: We had issues with RN50 tests at NVIDIA, getting `TypeError: 'list' object is not callable` failures, adding the shim that Keras repo uses fixed it.

Comment (maxhgerlach) [reply horovod/_keras/__init__.py:88]: thanks for explaining the context 👍

**Comment by EnricoMi** [comment]
As long as the link is valid, there is nothing we can do about. I reran the build jobs and the server seems to be fixed.

**Pull request merged**